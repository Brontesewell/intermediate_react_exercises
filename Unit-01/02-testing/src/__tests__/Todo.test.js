import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Todo from '../Todo';
import TodoForm from '../TodoForm';
import toJson from 'enzyme-to-json';

let props = {
  id: 1,
  handleDelete() {},
  handleEdit() {},
  toggleComplete() {},
  toggleEditForm() {}
};

it('renders without crashing', () => {
  shallow(<Todo {...props} />);
});

it('matches snapshot when complete', () => {
  const tree = renderer.create(<Todo {...props} isComplete={true} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('matches snapshot when incomplete', () => {
  const tree = renderer.create(<Todo {...props} isComplete={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('shows an edit form when editing', () => {
  const wrapper = shallow(<Todo {...props} isShowingEditForm={true} />);
  expect(wrapper.find(TodoForm)).toHaveLength(1);
});
