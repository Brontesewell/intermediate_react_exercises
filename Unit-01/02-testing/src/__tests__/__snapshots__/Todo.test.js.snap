exports[`test matches snapshot when complete 1`] = `
<div
  className="Todo complete">
  <h3>
    
  </h3>
  <p>
    
  </p>
  <div
    className="button-wrapper">
    <button
      className="complete-button"
      onClick={[Function]}>
      Mark as 
      incomplete
    </button>
    <button
      className="remove-button"
      onClick={[Function]}>
      Delete this todo
    </button>
    <button
      className="edit-button"
      onClick={[Function]}>
      Edit this todo
    </button>
  </div>
</div>
`;

exports[`test matches snapshot when incomplete 1`] = `
<div
  className="Todo ">
  <h3>
    
  </h3>
  <p>
    
  </p>
  <div
    className="button-wrapper">
    <button
      className="complete-button"
      onClick={[Function]}>
      Mark as 
      complete
    </button>
    <button
      className="remove-button"
      onClick={[Function]}>
      Delete this todo
    </button>
    <button
      className="edit-button"
      onClick={[Function]}>
      Edit this todo
    </button>
  </div>
</div>
`;
